### DISCUSSION TITLE
Is it possible to get square format for thumbnails ?

### DISCUSSION THREAD

**Discussion by creharmony** [igordanchenko/yet-another-react-lightbox#241] [Q&A] [answered]
Hi here, 

I'm using "yet-another-react-lightbox" + inline & thumbnails plugins.

I have multiple images with different formats : square format + 16/9 format images

With default sample I got "by height" thumbnails component, so
- 16/9 format images : fit well
- square format images: I got "hole" on the left and right of the image to fit the placeholder.

I would like to know if it's possible to switch on "square" thumbnails mode :
- square format image will fit the placeholder, 
- but for the 16/9 format image I want a square crop of the middle of the image (!hard?).

And as often an example with image (and with code) is worth 1000 words, I took up this question on stackblitz fork to illustrate.
[stackblitz sample here](https://stackblitz.com/edit/yet-another-react-lightbox-examples-fdm3ru?file=src%2Fexamples%2FInlinePlugin.tsx) => please go to "inline" example

any hint is appreciated.
NB: thanks for this well designed component + excellent doc !

**Comment by igordanchenko** [comment, marked as answer]
Hi there!

By default, the thumbnails (as well as image slides) are rendered with `imageFit: "contain"` setting. You can change it to `imageFit: "cover"` as follows.

```js
thumbnails={{
  position: "end",
  imageFit: "cover",
}}
```

Please let me know if this addresses your use case.

**Reply by creharmony** [reply]
with imageFit: cover 
- 16/9 format images : fit well
- square format image are cropped in the middle of the image height

that's not exactly my initial requirement. anyway this configuration is not that bad and may be a solution too 👍 thanks for you quick and efficient answer/support 🚀 

NB: I updated initial [stackblitz sample](https://stackblitz.com/edit/yet-another-react-lightbox-examples-fdm3ru?file=src%2Fexamples%2FInlinePlugin.tsx) with a new § related to your answer and will mark as answer your reply

**Reply by igordanchenko** [reply]
You can make the thumbnails square in addition to `imageFit: "cover"`:

```js
thumbnails={{
  position: "end",
  imageFit: "cover",
  width: 80,
  height: 80,
}}
```

**Reply by creharmony** [reply]
NB just for info: my side in global css I had an entry that pollute button render (I had to removed it to make it work)
````css
button {
  width: auto !important;
}
````

✅ initial [stackblitz sample](https://stackblitz.com/edit/yet-another-react-lightbox-examples-fdm3ru?file=src%2Fexamples%2FInlinePlugin.tsx) update with a final § 
❤️  huge thanks that's all fine now :)
